extension/property_tree/include/property_tree/string_parser.hpp

```diff
@@ -0,0 +1,147 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2023, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_PUBLIC_EXT_PROPERTY_TREE_STRING_PARSER_HPP_
+#define GKO_PUBLIC_EXT_PROPERTY_TREE_STRING_PARSER_HPP_
+
+
+#include <exception>
+#include <list>
+#include <string>
+#include <type_traits>
+
+
+#include <ginkgo/core/config/property_tree.hpp>
+
+
+namespace gko {
+namespace extension {
+
+
+std::vector<std::string> split_string(std::string str, char key = ' ')
+{
+    std::vector<std::string> vec;
+    auto sep = str.find(key);
+    while (sep != std::string::npos) {
+        if (sep != 0) {
+            vec.push_back(str.substr(0, sep));
+        }
+        str = str.substr(sep + 1);
+        sep = str.find(key);
+    }
+    if (str != "") {
+        vec.push_back(str);
+    }
+    return std::move(vec);
+}
+
+void string_parser(gko::config::pnode& ptree,
```

Comment:
This format requires type inference (is it bool, int, float, string, ...), which makes the parsing quite complex. Do we even need it?